src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -229,9 +230,19 @@ final case class ClientApiServiceImpl(
     val operationLabel: String = s"Deleting Key $keyId of Client $clientId"
     logger.info(operationLabel)
 
+    def checkRole(client: PersistentClient): Future[Unit] = for {
+      roles           <- getUserRolesFuture(contexts)
+      requesterUserId <- getUidFutureUUID(contexts)
+      _               <-
+        if (roles.contains(SECURITY_ROLE) && client.users.filter(_ == requesterUserId).isEmpty)
+          Future.failed(KeyOperationNotAllowedOnClient(client.id))
+        else Future.unit
+    } yield ()
```

Comment:
note: even admins cannot perform operations on a client where they are not members, we can skip role check

```suggestion
    def assertUserIsMemberOfClient(client: PersistentClient): Future[Unit] = for {
      requesterUserId <- getUidFutureUUID(contexts)
      _               <-
        if (!client.users.contains(requesterUserId))
          Future.failed(KeyOperationNotAllowedOnClient(client.id))
        else Future.unit
    } yield ()
```